### ISSUE TITLE
CSE Machine: error when assigning "undefined" to a variable.

### ISSUE THREAD

**Issue by CZX123** [source-academy/frontend#2781]
While running the below code:
```js
const x = undefined;
```
The CSE machine will get stuck at the step where `undefined` is being assigned to `x`. All subsequent steps will remain stuck on the same screen, but steps before this are still fine.
If you attempt to skip this assignment step using the slider, the steps after still will not work, but instead show whatever previous screen you were on.

Can be reproduced on https://sourceacademy.org/playground but works correctly on https://sourceacademy.nus.edu.sg/playground.

**Comment by RichDom2185** [comment]
https://sourceacademy.nus.edu.sg/ is frozen on commit e1a3267e5c4b1f63c067dd42d76e4533c70b8a55, so if it works on the latter, means the regression is likely to be somewhere between these commits: https://github.com/source-academy/frontend/compare/e1a3267e5c4b1f63c067dd42d76e4533c70b8a55...master

**Comment by CZX123** [comment]
I've found the reason [here](https://github.com/source-academy/frontend/commit/4b5911b393197bf97d1720c75783c003e9ba8081#diff-7dca6c18b97ffe7f765be0ad684bff9fbd6f80e6d4280aef393114fbb446a07cR117). The value of `this.environment.head[key]` is `undefined`, so the simplest fix would be to just place a `?` in front of the `descriptor` property.

However, what is more concerning is that any error that occurs are not thrown at all inside the browser console, as it did take me some time to debug this manually. I found out that all errors related to the CSE Machine are actually suppressed by the try-catch statements inside [JsSlangHelper.ts](https://github.com/source-academy/frontend/blob/master/src/commons/utils/JsSlangHelper.ts#L93) and [WorkspaceSaga.ts](https://github.com/source-academy/frontend/blob/master/src/commons/sagas/WorkspaceSaga.ts#L600).

May I know if there is a good reason why these try-catch statements are necessary? As without them, this issue and many others could have been solved much quicker.
